packages/arb-token-bridge-ui/src/hooks/useCompleteMultiChainTransactions.ts

```diff
@@ -0,0 +1,331 @@
+import { useLatest } from 'react-use'
+import { useAccount } from 'wagmi'
+import useSWRImmutable from 'swr/immutable'
+
+import { ChainId, rpcURLs } from '../util/networks'
+import { StaticJsonRpcProvider } from '@ethersproject/providers'
+import { getWagmiChain } from '../util/wagmi/getWagmiChain'
+import { useCallback, useEffect, useState } from 'react'
+import { fetchWithdrawalList } from '../util/withdrawals/fetchWithdrawals'
+import { fetchDepositList } from '../util/deposits/fetchDeposits'
+import {
+  L2ToL1EventResultPlus,
+  WithdrawalInitiated
+} from './arbTokenBridge.types'
+import { Transaction } from './useTransactions'
+import { MergedTransaction } from '../state/app/state'
+import {
+  getStandardizedTimestamp,
+  transformDeposit,
+  transformWithdrawal
+} from '../state/app/utils'
+import {
+  EthWithdrawal,
+  isTokenWithdrawal,
+  mapETHWithdrawalToL2ToL1EventResult,
+  mapTokenWithdrawalFromEventLogsToL2ToL1EventResult,
+  mapWithdrawalToL2ToL1EventResult
+} from '../util/withdrawals/helpers'
+import { FetchWithdrawalsFromSubgraphResult } from '../util/withdrawals/fetchWithdrawalsFromSubgraph'
+import { updateAdditionalDepositData } from '../util/deposits/helpers'
+
+const PAGE_SIZE = 100
+
+export type AdditionalProperties = {
+  direction: 'deposit' | 'withdrawal'
+  source: 'subgraph' | 'event_logs'
+  parentChainId: ChainId
+  chainId: ChainId
+}
+
+export type Deposit = Transaction
+
+export type Withdrawal = (
+  | FetchWithdrawalsFromSubgraphResult
+  | WithdrawalInitiated
+  | EthWithdrawal
+) &
+  AdditionalProperties
+
+type DepositOrWithdrawal = Deposit | Withdrawal
+
+function getStandardizedTimestampByTx(tx: DepositOrWithdrawal) {
+  if (isDeposit(tx)) {
+    return tx.timestampCreated ?? 0
+  }
+
+  if (isWithdrawalFromSubgraph(tx)) {
+    return getStandardizedTimestamp(tx.timestamp)
+  }
+
+  return getStandardizedTimestamp(tx.timestamp ?? '0')
+}
+
+function sortByTimestampDescending(
+  a: DepositOrWithdrawal,
+  b: DepositOrWithdrawal
+) {
+  return getStandardizedTimestampByTx(a) > getStandardizedTimestampByTx(b)
+    ? -1
+    : 1
+}
+
+const multiChainFetchList: { parentChain: ChainId; chain: ChainId }[] = [
+  {
+    parentChain: ChainId.Mainnet,
```

Comment:
I'd advocate for the data structure to have `sourceChain`, `destinationChain` instead of `parent/child` chain.
- parent-child relationship might not hold true for Orbit-Orbit or One-Nova transactions
- using source-destination relationship will automatically give a sense of direction. If source is parent, then it's a deposit, else it's a withdrawal.
- In the new bridge refactor PR, we are using terms like `sourceChain<Tx,Hash,Provider>`, `destinationChain<Tx,Hash,Provider>` so might as well keep it consistent right from the start.

Reply:
I agree with the notion of same layer chains not having a parent-child relationship, but src and dest chains won't work here. This is because we don't want to list deposits and withdrawals separately, so like:

```
src: Mainnet
dest: ArbOne

and then...

src: ArbOne
dest: Mainnet
```

So essentially we double the data for nothing. I will try to think of better names, open to more suggestions too.

Reply:
I'm thinking maybe we should leave it as it is for now. It's because we will have to refine what is deposit or withdrawal when fetching data. For example, if you make a transfer between Orbit chains, that's neither deposit or withdrawal, so subgraph/event logs will have to be restructured a bit anyway

Reply:
Okay, we might need to rethink this when this and Bridge-sdk code come together. I would need to keep a mental note for this.